### PR TITLE
Sync with PQClean

### PR DESCRIPTION
**Pull request by dstebila**
Sync with recent updates to PQClean, including NTRU AVX2 implementation.

* [No] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [No] Does this PR change the the list of algorithms available -- either adding or removing?  (If so, PRs in OQS-OpenSSL, OQS-BoringSSL, and OQS-OpenSSH will also be required by the time this is merged.)

### CODE REVIEW

**Comment by dstebila** [comment]
I'm not sure how to fix the error linking NTRU AVX2 code in the shared library setting. As far as I can tell, we do have -fPIC enabled.  Any ideas? @xvzcf @jschanck

File: src/kem/ntru/pqclean_ntruhrss701_avx2/poly_s3_to_rq.s [hunk 1]
```diff
@@ -0,0 +1,3118 @@
...
+vpand %ymm15, %ymm3, %ymm14
+vpxor %ymm14, %ymm2, %ymm3
+vpsrlq $1, %ymm3, %ymm2
+vpsubw %ymm2, %ymm0, %ymm2
+vpand const_modq(%rip), %ymm2, %ymm2
+vpand const_1s(%rip), %ymm3, %ymm3
+vpor %ymm3, %ymm2, %ymm3
+vmovdqa %ymm3, 1376(%rsp)
+mov %rsp, %rsi
+call PQCLEAN_NTRUHRSS701_AVX2_poly_Rq_mul_x_minus_1
```
Comment (jschanck) [inline src/kem/ntru/pqclean_ntruhrss701_avx2/poly_s3_to_rq.s:3116]: Changing this to
`call PQCLEAN_NTRUHRSS701_AVX2_poly_Rq_mul_x_minus_1@plt`
should fix the PIC issue.

Comment (dstebila) [reply src/kem/ntru/pqclean_ntruhrss701_avx2/poly_s3_to_rq.s:3116]: Yes, it seems to fix it for me on the gcc7-shared build. Can you make the commit at PQClean and then I'll pull it from there?

**Comment by dstebila** [comment]
Rebase once #819 lands

**Pull request merged**